tools/olm-bundle/README.md

```diff
@@ -0,0 +1,17 @@
+# OLM bundle generator
+Bash script for generating OLM bundle for OperatorHub.
+
+## Requirements
+- bash 5+ (GNU)
+- yq 4.6+ (YAML processor)
+- operator-sdk 1.19+
+
+## Usage
+```
+./generate-olm-bundle.sh <BUNDLE_VERSION>
```

Comment:
What is bundle version? Is that the same as operator version?

Reply:
TLDR; The bundle version is the version of the operator metadata package used by OLM. While OperatorHub uses the `package manifest` format[1], the `<BUNDLE_VERSION>` will be the same as the operator version. However, once OperatorHub moves to the `bundle` format, the `<BUNDLE_VERSION>` will be different from the operator version.

OperatorHub currently uses the `package manifest` format to release operator metadata packages, CRDs + CSV files, for OLM installation. The `package manifest` format does not require strict versioning of operator metadata packages so we can make updates to operator metadata of an operator version without versioning the operator metadata package. So while OperatorHub uses the `package manifest` format, we can keep this <BUNDLE_VERSION> as the same as the Operator version.

OLM has been moving towards using a new metadata package format called the `bundle` format, which is what OperatorHub will use in the future. The `bundle` format requires strict versioning of operator metadata packages, so whenever we update the operator metadata of an operator version, we must bump the operator metadata package version. This will force us to decouple the operator version from the operator metadata package version. For example, in the future we may have an operator of version `0.23.0` but then have an operator metadata package of version `0.23.0+0`

[1] https://docs.openshift.com/container-platform/4.7/operators/understanding/olm-packaging-format.html#olm-package-manifest-format_olm-packaging-format

Reply:
That is fine and I understand that. But I do not think this is clear from the readme. Especially the example with version `2.3.0` is confusing. Because in reality, even when the versioning is not exactly the same, it will still follow the operator versioning, right? It will not be _random-number+N_ but _operator-version+N_ for example. So having the example more reaslistic and setting it for example to 0.22.1 instead of 2.3.0 might make it more clear.

Reply:
Yes you are right, sorry, I meant to put `0.23.0` there but accidentally put `2.3.0`. I have just updated the example to `0.22.1`.